[2025-04-21T14:54:29.643+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-04-21T14:54:29.677+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: perbaikan_skripsi.predict_and_store_result scheduled__2025-04-21T14:51:00+00:00 [queued]>
[2025-04-21T14:54:29.707+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: perbaikan_skripsi.predict_and_store_result scheduled__2025-04-21T14:51:00+00:00 [queued]>
[2025-04-21T14:54:29.711+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 4
[2025-04-21T14:54:29.756+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): predict_and_store_result> on 2025-04-21 14:51:00+00:00
[2025-04-21T14:54:29.807+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'perbaikan_skripsi', 'predict_and_store_result', 'scheduled__2025-04-21T14:51:00+00:00', '--job-id', '353', '--raw', '--subdir', 'DAGS_FOLDER/tugas_akhir.py', '--cfg-path', '/tmp/tmpq4tf4og6']
[2025-04-21T14:54:29.810+0000] {standard_task_runner.py:105} INFO - Job 353: Subtask predict_and_store_result
[2025-04-21T14:54:29.807+0000] {logging_mixin.py:190} WARNING - /home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=30438) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-04-21T14:54:29.812+0000] {standard_task_runner.py:72} INFO - Started process 30507 to run task
[2025-04-21T14:54:29.989+0000] {task_command.py:467} INFO - Running <TaskInstance: perbaikan_skripsi.predict_and_store_result scheduled__2025-04-21T14:51:00+00:00 [running]> on host 1daba33e39bb
[2025-04-21T14:54:30.311+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Dion Rikki' AIRFLOW_CTX_DAG_ID='perbaikan_skripsi' AIRFLOW_CTX_TASK_ID='predict_and_store_result' AIRFLOW_CTX_EXECUTION_DATE='2025-04-21T14:51:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-04-21T14:51:00+00:00'
[2025-04-21T14:54:30.316+0000] {logging_mixin.py:190} INFO - Task instance is in running state
[2025-04-21T14:54:30.319+0000] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2025-04-21T14:54:30.321+0000] {logging_mixin.py:190} INFO - Current task name:predict_and_store_result state:running start_date:2025-04-21 14:54:29.680028+00:00
[2025-04-21T14:54:30.322+0000] {logging_mixin.py:190} INFO - Dag name:perbaikan_skripsi and current dag run status:running
[2025-04-21T14:54:30.325+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-04-21T14:54:30.329+0000] {logging_mixin.py:190} INFO - üîç [2] Memproses data dan melakukan prediksi...
[2025-04-21T14:54:30.422+0000] {logging_mixin.py:190} INFO - ‚úÖ Data siap diproses: 1 baris
   id  Temperature  Humidity  ...  nitrogen phosphorous potassium
0  88         25.3      84.0  ...       0.0         0.0       0.0

[1 rows x 9 columns]
[2025-04-21T14:54:30.425+0000] {logging_mixin.py:190} INFO - üìù Menyesuaikan nama kolom dengan model training...
[2025-04-21T14:54:30.428+0000] {logging_mixin.py:190} INFO - üîß Melakukan encoding kategori Soil_Type dan Crop_Type...
[2025-04-21T14:54:30.435+0000] {logging_mixin.py:190} INFO - üß† Menyiapkan fitur untuk prediksi...
[2025-04-21T14:54:30.438+0000] {logging_mixin.py:190} INFO - ü§ñ Menjalankan model prediksi...
[2025-04-21T14:54:30.470+0000] {logging_mixin.py:190} INFO - ‚úÖ Prediksi selesai:
   id predicted_fertilizer
0  88             10-10-10
[2025-04-21T14:54:30.472+0000] {logging_mixin.py:190} INFO - üíæ Menyimpan hasil ke tabel fertilizer_predictions...
[2025-04-21T14:54:30.543+0000] {logging_mixin.py:190} INFO - ‚ùå Error saat menyimpan ke database: (pymysql.err.IntegrityError) (1062, "Duplicate entry '88' for key 'fertilizer_predictions.sensor_data_id'")
[SQL: INSERT INTO fertilizer_predictions (sensor_data_id, predicted_fertilizer, created_at) VALUES (%(sensor_data_id)s, %(predicted_fertilizer)s, %(created_at)s)]
[parameters: {'sensor_data_id': 88, 'predicted_fertilizer': '10-10-10', 'created_at': datetime.datetime(2025, 4, 21, 14, 54, 30, 475900)}]
(Background on this error at: https://sqlalche.me/e/14/gkpj)
[2025-04-21T14:54:30.546+0000] {python.py:240} INFO - Done. Returned value was: None
[2025-04-21T14:54:30.572+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-04-21T14:54:30.574+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=perbaikan_skripsi, task_id=predict_and_store_result, run_id=scheduled__2025-04-21T14:51:00+00:00, execution_date=20250421T145100, start_date=20250421T145429, end_date=20250421T145430
[2025-04-21T14:54:30.779+0000] {logging_mixin.py:190} INFO - Task instance in success state
[2025-04-21T14:54:30.780+0000] {logging_mixin.py:190} INFO -  Previous state of the Task instance: running
[2025-04-21T14:54:30.781+0000] {logging_mixin.py:190} INFO - Dag name:perbaikan_skripsi queued_at:2025-04-21 14:52:00.824650+00:00
[2025-04-21T14:54:30.781+0000] {logging_mixin.py:190} INFO - Task hostname:1daba33e39bb operator:PythonOperator
[2025-04-21T14:54:30.832+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-04-21T14:54:30.875+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-04-21T14:54:30.879+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
